dev_docs/key_concepts/encrypted_saved_objects.mdx

```diff
@@ -0,0 +1,194 @@
+properties of an attribute are all included in its value. In this way, AAD inclusion is hierarchical.
+
+#### What attributes should be included in AAD
+
+Determining which attributes to include in AAD is not an exact science, however there are some basic guidelines.
+
+Good candidates for attributes to INCLUDE in AAD are attributes that...
+- have some association or relationship with an encrypted attribute (e.g. a configuration element of an ESO that reflects the shape of the encrypted data, the token type of a connector, the URL or TLS certificate for a monitor)
+- have a value that will never change once an object is created (e.g the created date or created by user, the type of action or connector)
+
+Good candidates for attributes to EXCLUDE from AAD are attributes that...
+
+- the value can be changed by an end user, and is meant to be updated separately from encrypted data, and has no association to any of the encrypted or AAD attributes (e.g. the name or UI properties of an object, the email title for an action or alert)
+- may not be present or populated, or populated algorithmically (e.g. any optional attributes, or calculated attributes like statistics or last updated time)
+- may be removed from the object or refactored in the future (e.g. deprecated or soon-to-be deprecated attributes, experimental attributes)
+- contain a large amount of data that can significantly slow down encryption and decryption, especially during bulk operations (e.g. large geo shape, arbitrary HTML document or image data)
+
+There are additional considerations to make due to how version upgrades work in Serverless. These are covered in more detail in the
+<DocLink id="kibDevDocsEncryptedSavedObjectsIntro" section="serverless-considerations" text="Serverless Considerations"/> section, but the basics are:
+
+- An attribute cannot be removed from AAD once it is included, unless it can be altogether removed from the object type, or refactored with a new name.
+- An existing attribute cannot be added to AAD if it was not included in AAD when it was first defined and has already been populated.
+
+When making the decision of which attributes to include in AAD, it is best to be conservative and only include attributes that the owning team is 100% confident
+should be included.
+
+## Caveats
+
+### Partial Updates
+
+Partial updates on ESOs are only possible if the changes are limited to unencrypted and non-AAD attributes. Any changes to an ESO's encrypted values or AAD-
+included values requires re-encryption, which means the entire object must be provided when updating to avoid corrupting the object. If an ESO is corrupted by
+a partial update, it will be effectively undecryptable. Currently, there is nothing preventing or limiting partial updates of ESOs (see open GitHub issue
+[50256](https://github.com/elastic/kibana/issues/50256)), so this requires consistent diligence from developers utilizing ESOs.
```

Comment:
(unrelated to the PR) TIL. I'll reply on the linked issue to reopen the discussion. Things have changed since 2019, we're now doing client-side updates, meaning that we do fetch the attributes from ES for partial updates, so this should be addressable (but maybe it requires encryption to be a first class citizen of the SOR first/\.